bentoml/deployment/sagemaker/__init__.py

```diff
@@ -0,0 +1,286 @@
+        os.chmod(os.path.join(snapshot_path, 'serve'), octal_permission)
+
+        execution_role_arn = get_arn_role_from_current_user()
+        ecr_image_path = create_push_image_to_ecr(self.bento_service, snapshot_path)
+        sagemaker_model_info = {
+            "ModelName": self.model_name,
+            "PrimaryContainer": {
+                "ContainerHostname": self.model_name,
+                "Image": ecr_image_path,
+                "Environment": {
+                    "API_NAME": self.api.name
+                }
+            },
+            "ExecutionRoleArn": execution_role_arn,
+        }
+        logger.info('Creating sagemaker model %s', self.model_name)
+        create_model_response = self.sagemaker_client.create_model(**sagemaker_model_info)
+        logger.info('AWS create model response: %s', create_model_response)
+
+        production_variants = [{
+            "VariantName": self.bento_service.name,
+            "ModelName": self.model_name,
+            "InitialInstanceCount": self.instance_count,
+            "InstanceType": self.instant_type,
+        }]
+        logger.info('Creating sagemaker endpoint %s configuration', self.endpoint_config_name)
+        create_endpoint_config_response = self.sagemaker_client.create_endpoint_config(
+            EndpointConfigName=self.endpoint_config_name, ProductionVariants=production_variants)
+        logger.info('AWS create endpoint config response: %s', create_endpoint_config_response)
+        logger.info('Creating sagemaker endpoint %s', self.bento_service.name)
+        create_endpoint_response = self.sagemaker_client.create_endpoint(
+            EndpointName=self.bento_service.name, EndpointConfigName=self.endpoint_config_name)
+        logger.info('AWS create endpoint response: %s', create_endpoint_response)
+
+        # TODO: maybe wait for this endpoint from creating to running and then return
+        return snapshot_path
+
+    def check_status(self):
+        endpoint_status_response = self.sagemaker_client.describe_endpoint(
+            EndpointName=self.bento_service.name)
+        logger.info('AWS describe endpoint response: %s', endpoint_status_response)
+        endpoint_in_service = False
+        if endpoint_status_response['EndpointStatus'] == 'InService':
+            endpoint_in_service = True
+
+        status_message = '{service} is {status}'.format(
+            service=self.bento_service.name, status=endpoint_status_response['EndpointStatus'])
+        if endpoint_in_service is True:
+            status_message += '\nEndpoint ARN: ' + endpoint_status_response['EndpointArn']
+
+        return endpoint_in_service, status_message
+
+    def delete(self):
+        """Delete Sagemaker endpoint for the bentoml service.
+        It will also delete the model or the endpoint configuration.
+
+        return: Boolean, True if the deletion is successful
+        """
+        active_status, _ = self.check_status()
+        if not active_status:
```

Comment:
could be `if not self.check_status()[0]`

Reply:
done